### PR TITLE
8264423: G1: Rename full gc attribute table states 

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  please review this cleanup change suggested in PR#2760 that renames some code in the g1 full gc attribute table.

This is the description from the CR:

>  Since JDK-8253600 g1 full gc has its own (temporary) attribute table storing information that it needs for evacuation.
> 
> Currently the naming corresponds to the attributes in HeapRegion. This is somewhat confusing, as they do not completely match.
> 
> While discussing this in the review for JDK-8262068 we thought of changing this to more reflect the purpose:
> 
> I.e.
> closed -> not_marked_through (or "always live" or "always_marked" something similar to indicate that we do not need to mark through them)
> pinned -> not_compacted
> normal -> compacted (or just keep "normal" as it is some internal state) 

Some additional renaming has been performed.

Testing: tier1-5

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264423](https://bugs.openjdk.java.net/browse/JDK-8264423): G1: Rename full gc attribute table states

### Reviewers
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**) ⚠️ Review applies to 2047bc11cf09f74b719cb0dde0bcdd6e811faf3f
 * [Hamlin Li](https://openjdk.java.net/census#mli) (@Hamlin-Li - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3486/head:pull/3486` \
`$ git checkout pull/3486`

Update a local copy of the PR: \
`$ git checkout pull/3486` \
`$ git pull https://git.openjdk.java.net/jdk pull/3486/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3486`

View PR using the GUI difftool: \
`$ git pr show -t 3486`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3486.diff">https://git.openjdk.java.net/jdk/pull/3486.diff</a>

</details>

### CODE REVIEW

**Review by kstefanj — APPROVED** [review]

File: src/hotspot/share/gc/g1/g1FullGCHeapRegionAttr.hpp [hunk 1]
```diff
@@ -54,35 +54,22 @@ class G1FullGCHeapRegionAttr : public G1BiasedMappedArray<uint8_t> {
...
-    return get_by_index(idx) == Pinned;
+    return get_by_address(obj) == Compacted;
   }
 
-  void set_normal(uint idx) { set_by_index(idx, Normal); }
-
-  bool is_normal(HeapWord* obj) const {
-    assert(!is_invalid(obj), "not initialized yet");
-    return get_by_address(obj) == Normal;
+  bool is_compacted_or_marked_through(uint idx) const {
```
Comment (Hamlin-Li) [inline src/hotspot/share/gc/g1/g1FullGCHeapRegionAttr.hpp]: Should this be rename to is_compacted_or_not_marked_through?
As the condition is indeed == Compacted || == NotMarkedThrough.

**Comment by kstefanj** [comment]
I think `SkipMarking` is better than the other alternatives so far.

File: src/hotspot/share/gc/g1/g1FullGCHeapRegionAttr.hpp (hunk 1 above)
Comment (tschatzl) [reply src/hotspot/share/gc/g1/g1FullGCHeapRegionAttr.hpp]: Yes, that's true. It should actually be called `is_compacted_or_not_marked_through`.

Comment (tschatzl) [reply src/hotspot/share/gc/g1/g1FullGCHeapRegionAttr.hpp]: See the other comment from me about naming variants provided by now.

**Comment by Hamlin-Li** [comment]
> I think `SkipMarking` is better than the other alternatives so far.

agree!

**Comment by tschatzl** [comment]
Given the apparent consensus, I'll keep `SkipMarking`. Doing some testing before any integration.

**Review by Hamlin-Li — APPROVED** [review]
looks good.

**Comment by tschatzl** [comment]
Retesting looks good. Thanks @kstefanj @Hamlin-Li for your reviews.

**Pull request closed without merging**